pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -531,14 +532,15 @@ public CompletableFuture<Void> messageAcked(CommandAck ack) {
 
     //this method is for individual ack not carry the transaction
     private CompletableFuture<Long> individualAckNormal(CommandAck ack, Map<String, Long> properties) {
-        List<Position> positionsAcked = new ArrayList<>();
+        List<Pair<Consumer, Position>> positionsAcked = new ArrayList<>();
         long totalAckCount = 0;
         for (int i = 0; i < ack.getMessageIdsCount(); i++) {
             MessageIdData msgId = ack.getMessageIdAt(i);
             Position position;
-            long ackedCount = 0;
-            long batchSize = getBatchSize(msgId);
-            Consumer ackOwnerConsumer = getAckOwnerConsumer(msgId.getLedgerId(), msgId.getEntryId());
+            Pair<Consumer, Long> ackOwnerConsumerPair = getAckOwnerConsumer(msgId.getLedgerId(), msgId.getEntryId());
```

Comment:
The owner consumer may be null, it seems that the ack owner consumer is mainly used to calculate un-ack message count, could we add a null-check for it?

Reply:
https://github.com/apache/pulsar/pull/23072/files#diff-6bdf62fdf16d74a56f5f9c1f5a6bf30f4360ad7022c69ab8f90c90a267bc568eR753

It won't be null, and if no matching conumser is found, will return `this`

Reply:
Maybe we shouldn't use `this` as the owner broker if we can't find the ack position in the consumer pending ack cache. For example:

1. Someone didn't consume messages and ack a position directly.
2. The broker restart.

Reply:
Actually, I threw an exception at first code, however, it caused the test to fail.
https://github.com/apache/pulsar/blob/411f6973e85b0a6213e992386e1704f93d0aae42/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BatchMessageWithBatchIndexLevelTest.java#L227

It was introduced by #17003

So, I kept the same behavior as the current logic.

I think we can discuss the behavior here in detail later and optimize it. In short, that's not the point of this PR

Reply:
ok